src/pybind/mgr/cephadm/inventory.py

```diff
@@ -163,12 +163,15 @@ def __init__(self, mgr):
         self.daemons = {}   # type: Dict[str, Dict[str, orchestrator.DaemonDescription]]
         self.last_daemon_update = {}   # type: Dict[str, datetime.datetime]
         self.devices = {}              # type: Dict[str, List[inventory.Device]]
+        self.osdspec_previews = {}     # type: Dict[str, List[Dict[str, Any]]]
         self.networks = {}             # type: Dict[str, Dict[str, List[str]]]
         self.last_device_update = {}   # type: Dict[str, datetime.datetime]
         self.daemon_refresh_queue = [] # type: List[str]
         self.device_refresh_queue = [] # type: List[str]
+        self.osdspec_previews_refresh_queue = [] # type: List[str]
         self.daemon_config_deps = {}   # type: Dict[str, Dict[str, Dict[str,Any]]]
         self.last_host_check = {}      # type: Dict[str, datetime.datetime]
+        self.loading_osdspec_preview = {}  # type: Dict[str, bool]
```

Comment:
Hm a dict of bools is a set, right? Is there a need to extend this data structure in the future?

Reply:
This setting is host-based though.

``` python
{'host1': True,
 'host2': False}
```

Reply:
yeah, that would be a set of host names.

Reply:
Not sure where you're getting at..

Reply:
```python
myset: Set[str] = set()
hostname in mysest

# and

mydict: Dict[str, bool] = {}
mydict[host]
```
is basically the same thing. With the exception that maintaining  the set is a bit easier.

Reply:
done